spec/04-basic-declarations-and-definitions.md

```diff
@@ -586,8 +586,8 @@ Def                ::=  ‘def’ FunDef
 FunDef             ::=  FunSig [‘:’ Type] ‘=’ Expr
 FunSig             ::=  id [FunTypeParamClause] ParamClauses
 FunTypeParamClause ::=  ‘[’ TypeParam {‘,’ TypeParam} ‘]’
-ParamClauses       ::=  {ParamClause} [[nl] ‘(’ ‘implicit’ Params ‘)’]
-ParamClause        ::=  [nl] ‘(’ [Params] ‘)’}
+ParamClauses       ::=  {ParamClause} [[nl] ‘(’ ‘implicit’ Params [‘,’] ‘)’]
+ParamClause        ::=  [nl] ‘(’ [Params] [‘,’] ‘)’}
```

Comment:
If you put trailing comma in the definition of `Params`, you wouldn't need to duplicate it everywhere. 

```
Params             ::=  Param {‘,’ Param} [‘,‘]
```